go/chat/utils/utils.go

```diff
@@ -344,19 +344,35 @@ func GetUnverifiedConv(ctx context.Context, g *libkb.GlobalContext, uid gregor1.
 	return inbox.ConvsUnverified[0], ratelim, nil
 }
 
-func FilterByType(msgs []chat1.MessageUnboxed, query *chat1.GetThreadQuery) (res []chat1.MessageUnboxed) {
-	if query != nil && len(query.MessageTypes) > 0 {
-		typmap := make(map[chat1.MessageType]bool)
+// FilterByType filters messages based on a query.
+// If includeAllErrors then MessageUnboxedError are all returned. Otherwise, they are filtered based on type.
+// Messages whose type cannot be determined are considered errors.
+func FilterByType(msgs []chat1.MessageUnboxed, query *chat1.GetThreadQuery, includeAllErrors bool) (res []chat1.MessageUnboxed) {
```

Comment:
Squicky gut reaction to boolean parameters, especially since this one has no existing callers for `false`. Your call.